torch/csrc/jit/codegen/cuda/lower_validation.cpp

```diff
@@ -88,8 +88,78 @@ void validateIr(Fusion* fusion) {
 
 namespace {
 
+// Check contiguity for all root domains associated with Misaligned Vectorize ParallelType
+void checkContiguity(
+    const std::unordered_set<IterDomain*>& domains,
+    TensorView* tv) {
+  TORCH_INTERNAL_ASSERT(tv->getMemoryType() == MemoryType::Global);
+
+  for (size_t idx = 0; idx < tv->getRootDomain().size(); ++idx) {
+    auto root = tv->getRootDomain()[idx];
+    if (domains.find(root) != domains.end()) {
+      TORCH_INTERNAL_ASSERT(
+          !root->isBroadcast(),
+          "Misaligned vectorization prohibits merging broadcast domains.",
+          "Issue found in, ",
+          tv);
+      TORCH_INTERNAL_ASSERT(
+          tv->domain()->contiguity()[idx],
+          "Cannot merge non-contiguous root domains with misaligned vectorization.",
+          "Issue found in, ",
+          tv);
+    }
+  }
+}
+
+// Check contiguity for all root domains associated with Misaligned Vectorize ParallelType
+void checkContiguity(
+    const std::unordered_set<IterDomain*>& domains,
+    TensorView* consumer,
+    TensorView* producer) {
+  TORCH_INTERNAL_ASSERT(consumer->getMemoryType() == MemoryType::Local);
+  TORCH_INTERNAL_ASSERT(producer->getMemoryType() == MemoryType::Global);
+
+  auto root_c2p =
+    PairwiseRootDomainMap(producer, consumer)
+      .mapConsumerToProducer(consumer->domain(), producer->domain());
+
+  std::unordered_map<IterDomain*, bool> producer_domain_contiguity;
+  for (size_t idx = 0; idx < producer->getRootDomain().size(); ++idx) {
+    auto root = producer->getRootDomain()[idx];
+    auto contiguity = producer->domain()->contiguity()[idx];
+    producer_domain_contiguity.insert({root, contiguity});
+  }
+
+  for (auto consumer_root : consumer->getRootDomain()) {
+    if (domains.find(consumer_root) != domains.end()) {
+      auto producer_root = root_c2p[consumer_root];
+      TORCH_INTERNAL_ASSERT(
+          producer_domain_contiguity.find(producer_root) !=
+          producer_domain_contiguity.end());
+
+      TORCH_INTERNAL_ASSERT(
+          !consumer_root->isBroadcast() || !producer_root->isBroadcast(),
+          "Misaligned vectorization prohibits merging broadcast domains.",
+          "Issue found in, ",
+          consumer);
```

Comment:
This isn't really checking contiguity. It should be done in a separate function, or the name of this function should be renamed (just `validateMisalignedVectoizationRootDomain` or something would be less confusing).

Reply:
I'll rename the function.